### PR TITLE
(test): Add ToS Allow List foundry tests

### PR DESCRIPTION
**Pull request by justinkaseman**
```
| File                                                                                   | % Lines           | % Statements      | % Branches      | % Funcs         |
| src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol                 | 96.00% (24/25)    | 96.00% (24/25)    | 90.00% (9/10)   | 100.00% (9/9)   |
```

### CODE REVIEW

**Comment by KuphJr** [comment]
Couple comments.  Otherwise it LGTM!

File: contracts/src/v0.8/functions/tests/1_0_0/FunctionsTermsOfServiceAllowList.t.sol [hunk 1]
```diff
@@ -1,52 +1,343 @@
...
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    vm.expectRevert("Only callable by owner");
+    s_termsOfServiceAllowList.unblockSender(STRANGER_ADDRESS);
+  }
+
+  event UnblockedAccess(address user);
+
+  function test_UnblockSender_Success() public {
```
[the remark this answers is not in the document]
Comment (justinkaseman) [reply contracts/src/v0.8/functions/tests/1_0_0/FunctionsTermsOfServiceAllowList.t.sol:322]: If they get unblocked? Yes

File: contracts/src/v0.8/functions/tests/1_0_0/FunctionsTermsOfServiceAllowList.t.sol [hunk 2]
```diff
@@ -1,52 +1,343 @@
...
+
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    assertEq(s_termsOfServiceAllowList.isBlockedSender(STRANGER_ADDRESS), true);
+  }
 }
 
 /// @notice #blockSender
```
[the remark this answers is not in the document]
Comment (justinkaseman) [reply contracts/src/v0.8/functions/tests/1_0_0/FunctionsTermsOfServiceAllowList.t.sol:268]: That scenario is covered by `test_BlockSender_Success` on line 281

**Review by KuphJr — DISMISSED** [review]

File: contracts/src/v0.8/functions/tests/1_0_0/FunctionsTermsOfServiceAllowList.t.sol [hunk 3]
```diff
@@ -1,52 +1,343 @@
...
+contract FunctionsTermsOfServiceAllowList_AcceptTermsOfService is FunctionsRoutesSetup {
+  function test_AcceptTermsOfService_RevertIfBlockedSender() public {
+    s_termsOfServiceAllowList.blockSender(STRANGER_ADDRESS);
+
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    bytes32 message = s_termsOfServiceAllowList.getMessage(STRANGER_ADDRESS, STRANGER_ADDRESS);
+    bytes32 prefixedMessage = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", message));
```
Comment (bolekk) [inline contracts/src/v0.8/functions/tests/1_0_0/FunctionsTermsOfServiceAllowList.t.sol:90]: put abi.encodePacked("\x19Ethereum Signed Message:\n32" in a contract-scoped var?

**Review by bolekk — APPROVED** [review]

**Review by KuphJr — APPROVED** [review]

**Pull request merged**